### PR TITLE
Timeframe options

### PR DESCRIPTION
**Pull request by swillner**
Following the discussion in #125 we need to give users options how their time series should be converted. This is a draft of how to implement such options in the `Timeframe` class.

- [ ] Implementation finished
- [ ] Tests added
- [ ] Documentation added
- [ ] Example added (in the documentation, to an existing notebook, or in a new notebook)
- [ ] Description in ``CHANGELOG.rst`` added (single line such as: ``(`#XX <https://github.com/openclimatedata/openscm/pull/XX>`_) Added feature which does something``)

### CODE REVIEW

**Comment by swillner** [comment]
@znicholls Just discussing the interpolation options in code here, right now I just altered the interface, not working yet in the background.

File: openscm/timeframes.py [hunk 1]
```diff
@@ -1,51 +1,107 @@
 """
 Different climate models often use different time frames for their input and output
-data. This includes different 'meanings' of time steps (e.g. beginning vs middle of year) and
-different lengths of the time steps (e.g. years vs months). Accordingly, OpenSCM
-supports the conversion of timeseries data between such timeframes, which is handled in
-this module. A thorough explaination of the procedure used is given in a dedicated
-`Jupyter Notebook
+data. This includes different 'meanings' of time steps (e.g. beginning vs middle of
```
Comment (znicholls) [inline openscm/timeframes.py]:
```suggestion
data. This includes different conventions for discrete timeseries data (e.g. beginning vs middle of
```

File: openscm/timeframes.py [hunk 2]
```diff
@@ -1,51 +1,107 @@
...
+    Interpolation type.
     """
 
+    LINEAR = 1
+    CUBIC = 3
+
+
+class TimeSeriesType(Enum):
+    """
+    Time series type.
```
Comment (znicholls) [inline openscm/timeframes.py]:
```suggestion
    Discrete to continuous conversion convention.
```

File: openscm/timeframes.py [hunk 3]
```diff
@@ -1,51 +1,107 @@
...
+    """
+
+    POINT = 1
```
Comment (znicholls) [inline openscm/timeframes.py]:
```suggestion
    POINT = 1  # OpenSCM currently only supports start of time period point data
```

File: openscm/timeframes.py [hunk 4]
```diff
@@ -1,51 +1,107 @@
...
     """
     Convenience class representing a timeframe consisting of a start time and a period
     length.
     """
 
-    start_time: int
-    """Start time (seconds since :literal:`1970-01-01 00:00:00`)"""
+    interpolation_type: InterpolationType
+    """
+    Type of interpolation to be used when a time series with this timeframe needs to be
```
Comment (znicholls) [inline openscm/timeframes.py]:
```suggestion
    Type of interpolation to be used to convert discrete data into continuous data when this timeframe needs to be
```

File: openscm/timeframes.py [hunk 5]
```diff
@@ -1,51 +1,107 @@
...
-    timeframe to another (e.g. when the target timeframe is outside the range of the
-    source timeframe) or when data is too short (less than 3 data points).
+    Interpolation type.
     """
 
+    LINEAR = 1
+    CUBIC = 3
+
+
+class TimeSeriesType(Enum):
```
Comment (znicholls) [inline openscm/timeframes.py]:
```suggestion
class DiscreteToContinousConvention(Enum):
```

File: openscm/timeframes.py [hunk 6]
```diff
@@ -1,51 +1,107 @@
...
+    converted.
+    """
+
+    points: Sequence[int]
+    """Time points (seconds since :literal:`1970-01-01 00:00:00`)"""
 
-    period_length: int
-    """Period length in seconds"""
+    time_series_type: TimeSeriesType
+    """Type of time series associated with this timeframe."""
```
Comment (znicholls) [inline openscm/timeframes.py]: Tweak re above to 'convention' language

**Comment by swillner** [comment]
Replaced by #133

**Pull request closed without merging**